autonomic-plugin-common/src/main/java/br/com/autonomiccs/autonomic/plugin/common/services/AutonomiccsSystemVmDeploymentService.java

```diff
@@ -234,32 +494,63 @@ private long getNextSystemVmId() {
         return id;
     }
 
+    /**
+     * <p>This method will look for the default network type for the given zone ({@link DataCenterVO}).
+     * If the given zone networking configuration is {@link DataCenterVO#getNetworkType()} equals to {@link NetworkType#Advanced} and {@link DataCenterVO#isSecurityGroupEnabled()} equals to true, we use the method {@link #getDefaultNetworkForAdvancedNetworkingWithSecurityGroupds(long, DataCenterVO)}.
+     * Otherwise, we use the method {@link #getDefaultNetwork(DataCenterVO)}.
+     *
+     * @param dataCenterId id of the zone
+     * @return the default network of the given zone
+     */
     private NetworkVO getDefaultNetwork(long dataCenterId) {
         DataCenterVO dc = dataCenterDao.findById(dataCenterId);
         if (dc.getNetworkType() == NetworkType.Advanced && dc.isSecurityGroupEnabled()) {
-            List<NetworkVO> networks = networkDao.listByZoneSecurityGroup(dataCenterId);
-            if (CollectionUtils.isEmpty(networks)) {
-                throw new CloudRuntimeException("Can not found security enabled network in SG Zone " + dc);
-            }
-            return networks.get(0);
-        } else {
-            TrafficType defaultTrafficType = TrafficType.Public;
-            if (dc.getNetworkType() == NetworkType.Basic || dc.isSecurityGroupEnabled()) {
-                defaultTrafficType = TrafficType.Guest;
-            }
-            List<NetworkVO> defaultNetworks = networkDao.listByZoneAndTrafficType(dataCenterId, defaultTrafficType);
+            return getDefaultNetworkForAdvancedNetworkingWithSecurityGroupds(dataCenterId, dc);
+        }
+        return getDefaultNetwork(dc);
+    }
 
-            // api should never allow this situation to happen
-            if (defaultNetworks.size() != 1) {
-                throw new CloudRuntimeException("Found " + defaultNetworks.size() + " networks of type " + defaultTrafficType + " when expect to find 1");
-            }
-            return defaultNetworks.get(0);
+    /**
+     * <p>This method will look for the default network type for the given zone ({@link DataCenterVO}).
+     * The method should be called only when network type meets either on of the following criterias:
```

Comment:
"_type meets either on of the following_"
to
"_either on**e** of the following_"